kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/slot-updates.k

```diff
@@ -0,0 +1,144 @@
+requires "word.md"
+
+module SLOT-UPDATES [symbolic]
+    imports EVM-TYPES
+    imports BUF
+    imports WORD
+
+    // ###########################################################################
+    // Auxiliaries
+    // ###########################################################################
+
+    // Index of first bit that equals one
+    syntax Int ::=  #getFirstOneBit(Int) [function, total]
+
+    rule [gfo-succ]: #getFirstOneBit(X:Int) => log2Int ( X &Int ( ( maxUInt256 xorInt X ) +Int 1 ) ) requires          #rangeUInt(256, X) andBool X =/=Int 0
+    rule [gfo-fail]: #getFirstOneBit(X:Int) => -1                                                    requires notBool (#rangeUInt(256, X) andBool X =/=Int 0)
+
+    // Index of first bit that equals zero
+    syntax Int ::= #getFirstZeroBit(Int) [function, total]
+
+    rule [gfz-succ]: #getFirstZeroBit(X:Int) => #getFirstOneBit ( maxUInt256 xorInt X ) requires         #rangeUInt(256, X)
+    rule [gfz-fail]: #getFirstZeroBit(X:Int) => -1                                      requires notBool #rangeUInt(256, X)
```

Comment:
```
#getFirstZeroBit(maxUInt256) =>
#getFirstOneBit(maxUInt256 xorInt maxUInt256) =>
#getFirstOneBit(0) =>
-1
```

Is this intended? I guess it makes sense, but just want to double-check with you the edge-case where we are in the "defined" case for `#getFirstZeroBit` but the "undefined" case for `#getFirstOneBit`.

Reply:
I wonder if we could have a test in `tests/specs/functional/slots-updates-spec.k` for cases like this, and we could add tests there as we discover more.

Reply:
Yes, this is intended.

Reply:
Sure, we could - what would you like to test, the behaviour of these auxiliary functions?

Reply:
Yeah, just on a few specific hardcoded values basically. Even concrete ones are fine I think, make sure they'r ebehaving as intended in `KClaim`s.

Reply:
Done now, to some extent.